staging/src/k8s.io/cloud-provider/app/controllermanager.go

```diff
@@ -289,6 +297,19 @@ func Run(c *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface
 	}
 
 	<-stopCh
+
+	// for graceful shutdown
+	if utilfeature.DefaultFeatureGate.Enabled(cmfeatures.CloudControllerManagerWebhook) {
+		if len(webhooks) > 0 {
+			<-webhookServerShutdownCh
```

Comment:
This is annoying that we have to match the if conditions and makes it more difficult to refactor. I wonder if we could do something with defer statements to achieve the same effect.

Reply:
Using the defer statement made the code more readable.
Thank you.
[use defer](https://github.com/kubernetes/kubernetes/pull/128888/commits/9cc325e79d67c64132431b93816a4b5d92c4a7d3)